src/main/groovy/com/ofg/uptodate/finder/MavenNewVersionFinderFactory.groovy

```diff
@@ -0,0 +1,20 @@
+package com.ofg.uptodate.finder
+
+import com.ofg.uptodate.LoggerProxy
+import com.ofg.uptodate.UptodatePluginExtension
+import groovy.transform.PackageScope
+
+@PackageScope
+class MavenNewVersionFinderFactory implements NewVersionFinderFactory {
+
+    final LoggerProxy loggerProxy
+
+    MavenNewVersionFinderFactory(LoggerProxy loggerProxy) {
+        this.loggerProxy = loggerProxy
+    }
+
+    @Override
+    NewVersionFinder build(UptodatePluginExtension uptodatePluginExtension) {
+        return new NewVersionFinder(loggerProxy, new MavenRepositorySettingsProvider(), new MavenLatestDependenciesProvider(loggerProxy), uptodatePluginExtension)
```

Comment:
we can pass configuration directly to the constructor instead of using settings provider